### PR TITLE
SAP Connector 5.2

### PR DESCRIPTION
**Pull request by dhirsch1**

### CODE REVIEW

File: sap/5.2/modules/ROOT/pages/sap-connector-reference.adoc [hunk 1]
```diff
@@ -0,0 +1,638 @@
...
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Field | Type | Description | Default Value | Required
+| Max Redelivery Count a| Number | The maximum number of times a message can be redelivered and processed unsuccessfully before triggering process-failed-message. |  |
+| Use Secure Hash a| Boolean | Whether to use a secure hash algorithm to identify a redelivered message. |  |
+| Message Digest Algorithm a| String | The secure hashing algorithm to use. If not set, the default is SHA-256. |  |
+| Id Expression a| String | Defines one or more expressions to use to determine when a message has been redelivered. This property may only be set if useSecureHash is `false`. |  |
+| Object Store a| Object Store | The object store where the redelivery counter for each message is stored. |  |
+|===
+
```
Comment (Santaionut) [inline sap/5.2/modules/ROOT/pages/sap-connector-reference.adoc:661]: please add here this :


[[IDocParseParameterGroup]]
=== I Doc Parse Parameter Group

[cols=".^20%,.^25%,.^30%,.^15%,.^10%", options="header"]
|======================
| Field | Type | Description | Default Value | Required
| Parse With Field Value Checking a| Boolean | Parsing option for enabling the field value checking mechanism in accordance with the respective IDoc field metadata. | false | 
| Parse Ignore Unknown Fields a| Boolean | Parsing option for ignoring unknown fields, e.g. when parsing with older IDoc metadata. | false | 
| Parse Ignore Invalid Char Errors a| Boolean | Parsing option for ignoring character range check errors in accordance with the declared XML specification version.
 If using this option invalid characters within field values are accepted although the XML document to parse may not be well-formed because it potentially contains characters not being in the valid character range. | false | 
| Parse Without Field Data Type Checking a| Boolean | Parsing option for disabling the field data type checking mechanism in accordance with the respective IDoc field metadata.
 Caution: This option should be used with care.
 Sending IDocs with illegal field contents that are violating the field's defined datatype may cause ABAP system dumps and result in unrecognized partial loss of data when later being processed in an AS ABAP system. | false | 
| Parse Refuse Unknown XML Version a| Boolean | Parsing option for refusing XML documents that declare to use an unknown XML specification version, i.e. a potential future XML specification version 1.2 or 2.0. | false | 
| Parse Refuse XML Version10 a| Boolean | Parsing option for refusing XML documents that declare to use the XML specification version 1.0 or do not contain an XML declaration header. | false | 
| Parse Refuse XML Version11 a| Boolean | Parsing option for refusing XML documents that declare to use the XML specification version 1.1. | false | 
| Parse Accept Only Xml Version10 a| Boolean | Parsing option for accepting only XML documents that declare to use the XML specification version 1.0 or do not contain an XML declaration header. | false | 
| Parse Accept Only Xml Version11 a| Boolean | Parsing option for accepting only XML documents that declare to use the XML specification version 1.1. | false | 
| Parse Accept Only Xml Version10 To11 a| Boolean | Parsing option for accepting only XML documents that declare to use the XML specification version 1.0 or 1.1 or do not contain an XML declaration header. | false | 
|======================

File: sap/5.2/modules/ROOT/pages/sap-connector-reference.adoc [hunk 2]
```diff
@@ -0,0 +1,638 @@
...
+[[SapAttributes]]
+=== SAP Attributes
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Field | Type | Description | Default Value | Required
+| Transaction Id a| String |Transaction ID value.  |  |
+|===
+
```
Comment (Santaionut) [inline sap/5.2/modules/ROOT/pages/sap-connector-reference.adoc:588]: please add this:

[[IDocRenderParameterGroup]]
=== I Doc Render Parameter Group

[cols=".^20%,.^25%,.^30%,.^15%,.^10%", options="header"]
|======================
| Field | Type | Description | Default Value | Required
| Render I Doc Empty Tags a| Boolean | Rendering option for including also all empty tags in the resulting XML document. | false | 
| Render I Doc Ignore Invalid Char Errors a| Boolean | Rendering option for ignoring character range check errors in accordance with the used XML specification version. | false | 
| Render I Doc Ignore Tag Escape Errors a| Boolean | Rendering option for ignoring escape character errors within XML tags in accordance with the used IDocXML format. | false | 
| Render I Doc Omit Xml Declaration a| Boolean | Rendering option for omitting the XML declaration section in the resulting XML document. | false | 
| Render I Doc Optional Encoding Decl a| Boolean | Rendering option for writing also an optional XML encoding declaration attribute into the resulting XML document. | false | 
| Render I Doc Substitute Invalid Chars a| Boolean | Rendering option for substituting invalid characters within IDoc field values with the number sign '#' (U+0023). | false | 
| Render I Doc With CR a| Boolean | Rendering option for inserting CR (Carriage Return) characters into the resulting XML document. | false | 
| Render I Doc With CRLF a| Boolean | Rendering option for inserting CR+LF (Carriage Return and Line Feed) characters into the resulting XML document. | false | 
| Render I Doc With LF a| Boolean | Rendering option for inserting LF (Line Feed) characters into the resulting XML document. | false | 
| Render I Doc With Tabs And CR a| Boolean | Rendering option for inserting TAB and CR (Tabulators and Carriage Return) characters into the resulting XML document. | false | 
| Render I Doc With Tabs And CRLF a| Boolean | Rendering option for inserting TAB and CR+LF (Tabulators, Carriage Return and Line Feed) characters into the resulting XML document. | false | 
| Render I Doc With Tabs And LF a| Boolean | Rendering option for inserting TAB and LF (Tabulators and Line Feed) characters into the resulting XML document. | false | 
| Render I Doc Xml10 a| Boolean | Rendering option for creating an XML document compliant to the XML specification version 1.0 | false | 
| Render I Doc Xml11 a| Boolean | Rendering option for creating an XML document compliant to the XML specification version 1.1 | false | 
|======================

File: sap/5.2/modules/ROOT/pages/sap-connector-reference.adoc [hunk 3]
```diff
@@ -0,0 +1,638 @@
...
+==== Parameters
+[%header,cols="20s,20a,35a,20a,5a"]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | The name of the configuration to use. | | x
+| Gateway Host a| String |  Name of the host running the gateway server. |  | x
+| Gateway Service a| String |  Either the name or service port for the gateway service of the SAP system. |  | x
+| Program ID a| String |  SAP system program ID that is registered on the gateway. Must be unique for the given gateway. |  | x
+| Connection Count a| Number |  The number of connections to register on the gateway. |  1 |
+| Idoc Type Filter Regex a| String |  Expression to filter by type, all incoming IDocs. |  |
```
Comment (Santaionut) [inline sap/5.2/modules/ROOT/pages/sap-connector-reference.adoc:128]: please add this below :

| IDoc Render Options a| <<`IDocRenderParameterGroup`>> |  +++IDoc Rendering options+++ |  |

File: sap/5.2/modules/ROOT/pages/sap-connector-reference.adoc [hunk 4]
```diff
@@ -0,0 +1,638 @@
...
+==== Parameters
+[%header,cols="20s,20a,35a,20a,5a"]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | The name of the configuration to use. | | x
+| IDoc Name a| String |  The name of the iDocument to execute. |  | x
+| Content a| Binary |  The IDocument to execute. |  #[payload] |
+| Version a| String |  The version on the IDoc. |  |
+| Transaction Id a| String |  The ID that identifies an RFC so it runs only once. |  |
+| Queue Name a| String |  Name of the queue on which the RFC executes. This applies for qRFC since each tRFC call that is processed is given a queue name by the application. A queue name and version are required for each qRFC transaction. |  |
```
Comment (Santaionut) [inline sap/5.2/modules/ROOT/pages/sap-connector-reference.adoc:398]: please add this below:

| IDoc Parsing Options a| <<`IDocParseParameterGroup`>> |  |  |

**Review by Santaionut — DISMISSED** [review]

File: sap/5.2/modules/ROOT/pages/sap-connector-examples.adoc [hunk 5]
```diff
@@ -0,0 +1,243 @@
+= SAP Connector 5.2 Examples - Mule 4
+:page-aliases: connectors::sap/sap-connector-examples.adoc
+
+== Establish a SAP Connection
```
Comment (valkyrie69) [inline sap/5.2/modules/ROOT/pages/sap-connector-examples.adoc]:
```suggestion
== Establish an SAP Connection
```

Comment (valkyrie69) [reply sap/5.2/modules/ROOT/pages/sap-connector-examples.adoc]: GLOBAL: SAP isn't "sap", it's Ess A Pee, so article needs to match.

File: sap/5.2/modules/ROOT/pages/sap-connector-studio.adoc [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+. Configure the SAP connector.
+. Configure an input source for the connector.
+
+[[add-connector-to-project]]
+== Add the Connector to Your Mule Project
+
+Add SAP Connector to your Mule project to automatically populate the XML code with the connector's namespace and schema location and to add the required dependencies to the project's `pom.xml` file:
+
+. In the *Mule Palette* view, click *(X) Search in Exchange*.
+. In the *Add Dependencies to Project* window, type `<connector name in lowercase>` in the search field.
```
Comment (valkyrie69) [inline sap/5.2/modules/ROOT/pages/sap-connector-studio.adoc]:
```suggestion
. In the *Add Dependencies to Project* window, type `sap` in the search field.
```

File: sap/5.2/modules/ROOT/pages/sap-connector-studio.adoc [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+. Click *SP Connector* in *Available modules*.
+. Click *Add*.
+. Click *Finish*.
+
+Adding a connector to a Mule project in Studio does not make that connector available to other projects in your Studio workspace.
+
+== Select a Source Operation
+
+. In Anypoint Studio, click the *Mule Palette* and type "sap".
```
Comment (valkyrie69) [inline sap/5.2/modules/ROOT/pages/sap-connector-studio.adoc]:
```suggestion
. In Anypoint Studio, click the *Mule Palette* and type `sap`.
```

File: sap/5.2/modules/ROOT/pages/sap-connector-studio.adoc [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+=== Install the JCO Native Library
+
+. Click *Add Dependency*.
+. Click *Install*.
++
+image::sap-connector-configuration-02.png[]
++
+. In the *Install* menu, click *Browse* to locate the library file.
++
+On MacOS, starting with JCO version 3.1.2, the native library extension was changed from `jnilib` to `dylib`. Anypoint Studio versions 7.0 through 7.6 have a limitation on selecting the `dylib` file type.
```
Comment (valkyrie69) [inline sap/5.2/modules/ROOT/pages/sap-connector-studio.adoc]: I think we need to maybe reword this, something like: Starting with Anypoint Studio 7.0, there is a limitation..."
Because Studio is now on version 7.7. Maybe this limitation doesn't apply to 7.7? Needs to be verified with dev.

File: sap/5.2/modules/ROOT/pages/sap-connector-studio.adoc [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+== Test the Connection
+
+. Sign into your SAP login to get the information you need to configure the remaining *Global Element Properties* fields as described in <<gepparms,Configure in Studio>>.
+. Click *Test Connection* to ensure that you have the correct login information.
+
+== Configure the Connector
+
+If you need more than one SAP Connector instance in your Mule project, you can create a global SAP element. The SAP Connector object holds the configuration properties that allow you to connect to the SAP server.
+
+To create a configuration for an SAP Connector, complete the following steps:
```
Comment (valkyrie69) [inline sap/5.2/modules/ROOT/pages/sap-connector-studio.adoc]:
```suggestion
To create a configuration for SAP Connector, complete the following steps:
```

File: sap/5.2/modules/ROOT/pages/sap-connector-reference.adoc [hunk 10]
```diff
@@ -0,0 +1,690 @@
+= SAP Connector 5.2 Reference - Mule 4
+:page-aliases: connectors::sap/sap-connector-reference.adoc
+
+Support Category: https://www.mulesoft.com/legal/versioning-back-support-policy#anypoint-connectors[Premium]
+
+Anypoint Connector for SAP (SAP Connector) enables the integration of data to and
+from external systems based on SAP NetWeaver.
+
+NOTE: Starting with version 5.0, SAP Connector also supports connecting to an on-premise instance of SAP S/4HANA.
```
Comment (valkyrie69) [inline sap/5.2/modules/ROOT/pages/sap-connector-reference.adoc]:
```suggestion
NOTE: Starting with version 5.0, SAP Connector also supports connecting to an on-premises instance of SAP S/4HANA.
```

**Review by valkyrie69 — APPROVED** [review]
